alerts/alerts.go

```diff
@@ -248,6 +259,29 @@ func (m *manager) Filter(alerts []*api.Alert, filters ...Filter) ([]*api.Alert,
 	return alerts, nil
 }
 
+func (m *manager) Purge() error {
+	if (m.useTtl) {
+		return nil
+	}
+	if kvdb.Instance() != nil {
+		kvps, err := enumerate(kvdb.Instance(), kvdbKey)
+		if err == nil {
+			for _, kvp := range kvps {
+				alert := new(api.Alert)
+				if err := json.Unmarshal(kvp.Value, alert); err != nil {
+					continue
+				}
+				curtime := (int64)(time.Now().Unix())
+				if (curtime - alert.Timestamp.GetSeconds()) > (int64)(alert.Ttl) {
+					//Alert has lived its ttl. Time to delete it.
+					kvdb.Instance().Delete(kvp.Key)
```

Comment:
I would say check for how many deletes fail and return an error in this function indicating some alerts were not purged.